### DISCUSSION TITLE
WinlogBeats - Missing Logs

### DISCUSSION THREAD

**Discussion by Git-Me-Some-Hub** [Security-Onion-Solutions/securityonion#6722] [Unsupported Versions]
ISO - 2.3.91 - Manager / Search / Forwarder

I am having the hardest time understanding why my production SO instance is not collecting all the Security Events while I can have it all work fine in the my Test environment. 

I recently first noticed it with the log clear id (1102), and now I am see it with 4662 as well. I am completely stumped on what my be happening. I used to install the Package from Elastic that matched up with SO, but was testing out the MSI file that I can download from SO Console. Still not working right. 

![image](https://user-images.githubusercontent.com/17930434/147694270-fb2e6838-fdb5-4734-8d05-2ab3c8d3f980.png)
![image](https://user-images.githubusercontent.com/17930434/147695123-26ab3566-12cd-40b0-84ab-a3b9a718db92.png)
  
I have tried to use the example file that is provided with the MSI, I have used the config file that is working in my Test environment, too, but nothing seems to help. Stumped.

**Comment by Git-Me-Some-Hub** [comment]
It seems the .winlogbeat.yml file is not matching the config file I put out there, but it is listing Security? 

![image](https://user-images.githubusercontent.com/17930434/147700821-87e3f7fc-b144-4fb7-8a8b-659b86b617e0.png)
![image](https://user-images.githubusercontent.com/17930434/147700847-00475fbe-f606-4887-9d70-f190549af0f3.png)

**Comment by dougburks** [comment]
We recommend using the SOC Downloads page:
https://docs.securityonion.net/en/latest/beats.html#winlogbeat

If you've previously installed a different version of Winlogbeat, have you tried fully removing the service and all configuration before installing the new version?

When searching in Kibana, is it possible that your query results are being limited? Have you tried filtering your search to limit the result set?

Have you tried searching in Hunt to see if it shows anything different?

**Reply by Git-Me-Some-Hub** [reply]
Understand, I am going to try and standardize on the correct version going forward; however, I am kind of stuck as is for right now and I switching seems not to help. I did the uninstall of the old service on my current laptop I am testing with, as well as another VM... Same result. 

Sorry I was meaning to get some Hunt shots, so you all knew it was not some Kibana stuff. 
Laptop: Cleared logs many times 
![image](https://user-images.githubusercontent.com/17930434/147982238-ebd54e9d-38f2-4067-a421-c7c70acccee5.png)

I broadened it out to all 1102 events, but it seem the two I see are in the Application log for an email server. 
![image](https://user-images.githubusercontent.com/17930434/147982464-0b951615-2e5e-4ccb-8f5f-1557ca4d9e66.png)

Here is the Kibana Search I did from my laptop and my test vm, I should have had a clear at 14:20
![image](https://user-images.githubusercontent.com/17930434/147982658-a4510b21-1b83-4b85-967a-6c8db951337a.png)
![image](https://user-images.githubusercontent.com/17930434/147982688-b2c9e82f-97d8-4c3c-9a96-3eeaec669463.png)

**Reply by defensivedepth** [reply]
@Git-Me-Some-Hub Please post the full winlogbeat config file that is currently in use - Feel free to drop it over on https://gist.github.com/

**Reply by Git-Me-Some-Hub** [reply]
@defensivedepth - Absolutely, 
https://gist.github.com/Git-Me-Some-Hub/04c0bccbcde67b77c166268aabcb5065

**Reply by Git-Me-Some-Hub** [reply]
I also went ahead and installed the Winlogbeats-oss version 7.16.2, no other version was installed previously. Still I am not seeing that specific event code come up.

**Reply by defensivedepth** [reply]
Are you seeing any errors from Logstash?

` sudo tail -f /opt/so/log/logstash/logstash.log`

**Reply by Git-Me-Some-Hub** [reply]
All that I see is: 
`[2022-01-05T17:55:55,196][WARN ][deprecation.logstash.codecs.plain] Relying on default value of `pipeline.ecs_compatibility`, which may change in a future major release of Logstash. To avoid unexpected changes when upgrading Logstash, please explicitly declare your desired ECS Compatibility mode.
`
But I am seeing this error in my Test environment, and have for some time.

**Reply by Git-Me-Some-Hub** [reply]
So my Test Environment is not a 1-1 match, and currently not really meant to be. I am shipping logs to my Manger, and I have one Search node after that. 

Just to confirm I have this right: 
Flow: Agents (beats) > Manger (elastic) > Search

**Reply by defensivedepth** [reply]
The flow is more like:

Agents (beats) > Manager (Logstash --> Elasticsearch Ingest Processors --> Elasticsearch) > Search

So to be clear - according to the WLB config you posted, WLB should only be sending eventid 1102 from the Security eventlog. That would be why no other logs another than 1102 are being sent.

**Reply by Git-Me-Some-Hub** [reply]
Right, sorry, yeah there is a lot more going on in the manager than that condense view. I was just making sure I was not doing some dumb mistake. 

As for the config, I guess I forgot that I tweaked it for testing. Even with this testing config I did not get the 1102 eventId to show up. Here is another config, should be nearly the exact same, sans the call out for that eventId. I have also used the default winlogbeat.example.yml from your installation as well. None of these have seem to fix it. 
https://gist.github.com/Git-Me-Some-Hub/4d857751dbfcead1833bf89fcf518658

Seems that you are probably at a loss of the cause of this issue as well. 

I did test it again with the example config that comes with the MSI install on a laptop that had never had the other winlogbeat installation. Even with the MSI being the only thing installed on this machine it did not change the result. no 1102 eentId.

**Comment by defensivedepth** [comment]
So when troubleshooting data flow issues like this, I typically take the approach of following the data flow itself and making sure it is being sent / received from start to finish. 

So on the the Windows system, check out the WLB log under:

`C:\ProgramData\Elastic\Beats\winlogbeat\logs\winlogbeat`

You should see something like this:

2022-01-06T10:30:10.439-0500	INFO	[publisher_pipeline_output]	pipeline/output.go:151	Connection to backoff(async(tcp://x.x.x.x:5044)) established
2022-01-06T10:30:11.511-0500	INFO	beater/eventlogger.go:88	EventLog[Microsoft-Windows-Sysmon/Operational] successfully published 1200 events
2022-01-06T10:30:11.514-0500	INFO	beater/eventlogger.go:88	EventLog[Security] successfully published 626 events
2022-01-06T10:30:11.514-0500	INFO	beater/eventlogger.go:88	EventLog[System] successfully published 162 events
2022-01-06T10:30:11.514-0500	INFO	beater/eventlogger.go:88	EventLog[Windows PowerShell] successfully published 16 events
2022-01-06T10:30:11.514-0500	INFO	beater/eventlogger.go:88	EventLog[Application] successfully published 7 events
2022-01-06T10:30:11.514-0500	INFO	beater/eventlogger.go:88	EventLog[Microsoft-Windows-PowerShell/Operational] successfully published 1 events
2022-01-06T10:30:11.875-0500	INFO	beater/eventlogger.go:88	EventLog[Microsoft-Windows-Sysmon/Operational] successfully published 1050 events

If you clear the security eventlog (which generates an 1102), can you verify that the WLB shows that it published 1 log from EventLog[Security]?

**Reply by Git-Me-Some-Hub** [reply]
Makes sense, I have seen those specific call outs in the past saying what has been sent; however, I am not really seeing them now, thought events are still being sent. I am seeing some errors there about the connection being forcibly terminated?   Added link below to match screenshots. 

Event Viewer: 
![image](https://user-images.githubusercontent.com/17930434/148411404-40895abb-780a-4f3b-9ece-5221ff60c361.png)
![image](https://user-images.githubusercontent.com/17930434/148411601-79b1ed0c-2716-407f-b63b-f9ac6eac8954.png)

Here is the link to the Logs, so it doe snot clog up this chat. 
https://gist.github.com/Git-Me-Some-Hub/0cc1f01f5d80fb0405615b788ea0ee47

**Reply by defensivedepth** [reply]
Can you drill down into one of those events in Kibana (one of the events pictured in screen cap) - what is the value of the `_index` field?

**Reply by Git-Me-Some-Hub** [reply]
Apologies, I have been neck deep in other stuff. Thanks for the continued help. The index is normal. Like that other post you and I responded to, it may be the Field limitation error, but I am not really sure how to properly implement that base on the guide you all list in documentation. Is it a query to do the change, or is it a config file? 

Index: 
![image](https://user-images.githubusercontent.com/17930434/151057087-6b0bc96e-38b5-4f3a-83ed-35eeef39ecf9.png)

**Reply by dougburks** [reply]
The documentation shows a temporary change (https://docs.securityonion.net/en/2.3/elasticsearch.html#temporary) and a persistent change (https://docs.securityonion.net/en/2.3/elasticsearch.html#persistent). From the persistent section, you'll most likely want both:
> Please note that the change to the field limit will not occur immediately, only on index creation. Therefore, it is recommended to run the previously mentioned temporary command and modify the template file.

**Reply by Git-Me-Some-Hub** [reply]
Thanks @dougburks, while I do understand doing this for a specific index in a temporary situation I am not quite sure I do for the other all template ? 

I did the temporary increase to the fields, I then cleared the Windows Event Logs but am not getting the expected 1102 Event reported. I did still see errors with the filed limit at 2000, so I went to 3000 and still no change. 

`curl -k -XPUT -H'Content-Type: application/json' -u <user>:<password> https://localhost:9200/so-beats-*/_settings -d'{ "index.mapping.total_fields.limit": 2000 }'`
`curl -k -XPUT -H'Content-Type: application/json' -u <user>:<password> https://localhost:9200/so-beats-2022.01.26/_settings -d'{ "index.mapping.total_fields.limit": 2000 }'`

![image](https://user-images.githubusercontent.com/17930434/151232169-8025629b-7db0-4e03-8c67-9cbbe8125357.png)
